File: MATTHEW/leetcode/AIcoding/4.GPT-4o/c_solutions/1422-maximum-score-after-splitting-a-string.c
int maxScore(char * s) {
    int score = 0, maxScore = 0, n = strlen(s);
    for (int i = 0; i < n; i++) {
        if (s[i] == '0') score++;
    }
    maxScore = score;
    for (int i = 0; i < n - 1; i++) {
        if (s[i] == '0') score--;
        else score++;
        if (score > maxScore) maxScore = score;
    }
    return maxScore;
}

File: leetcode/AIcoding/2.Gemini-2.0/erlang_solutions/1422-maximum-score-after-splitting-a-string.erl
-module(maximum_score_after_splitting_a_string).
-export([max_score/1]).

max_score(S) ->
    max_score(S, 0, 0, 0).

max_score([], Ones, Zeros, Max) ->
    Max;
max_score([H|T], Ones, Zeros, Max) ->
    NewZeros = case H of
                   $0 -> Zeros + 1;
                   _ -> Zeros
               end,
    NewOnes = case H of
                   $1 -> Ones + 1;
                   _ -> Ones
               end,
    Len = length(T),
    if Len > 0 ->
        Score = NewZeros + (lists:sum([1 || C <- T, C == $1])),
        NewMax = max(Max, Score),
        max_score(T, NewOnes, NewZeros, NewMax);
    true ->
        Max
    end.